cmd/ctl/pkg/check/api/api.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/jetstack/cert-manager/pkg/util/cmapichecker"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	restclient "k8s.io/client-go/rest"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// Options is a struct to support check api command
+type Options struct {
+	RESTConfig *restclient.Config
+
+	// APIChecker is used to check that the cert-manager CRDs have been installed on the K8S
+	// API server and that the cert-manager webhooks are all working
+	APIChecker cmapichecker.Interface
+
+	// If set to true, command will wait until creating resources against the api is possible
+	Wait bool
+
+	// Time before timeout when waiting
+	Timeout time.Duration
+
+	// Time between checks when waiting
+	Interval time.Duration
+
+	// Namespace that is used to dry-run create the certificate resource in
+	Namespace string
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// Complete takes the command arguments and factory and infers any remaining options.
+func (o *Options) Complete(factory cmdutil.Factory) error {
+	var err error
+
+	o.Namespace, _, err = factory.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig, err = factory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.APIChecker, err = cmapichecker.New(o.RESTConfig, o.Namespace)
+	if err != nil {
+		return err
```

Comment:
It'd be nice to wrap this error with some human readable context since it'll be printed to the user's console.

Reply:
I added `return fmt.Errorf("Error: %v", err)`, additionally the error will have had some context added in `cmapichecker.New`